### PR TITLE
feat: add logging to ts-sdk for ranger monitoring

### PR DESCRIPTION
**Pull request by petscheit**
# Summary of changes
Adds basic logging to enable the monitoring of the ranger on roco.

If it fixes a bug or creates a feature, link the issue.

## Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue):
    - [ ] Have you **written tests **to protect against future occurrences of the bug?
- [x] New feature (non-breaking change which adds functionality)
    - [x] Have you **seen it working** in a development environment?
    - [ ] Have you **written tests** for **happy** and **unhappy** paths?
    - [ ] Have you implemented this feature as per the **issue acceptance criteria**?
- [ ] Breaking change (a feature that would cause existing functionality not to work as expected
    - [ ] Is the breaking change **documented**?
    - [ ] Has any previous changes been **deprecated**?
- [ ] CI/CD
    - [ ] If introducing new actions, have you **looked at the action code** for anything spurious?
    - [ ] Have you written **custom scripts** for things that could be actions already on the marketplace?
    - [ ] If introducing new actions, have you **pegged a static version**?
- [ ] Documentation Update
    - [ ] Have you checked other documentation, such as the docs repository?
    - [ ] If updating script documentation, have you **tested it on both environments**?
- [ ] Substrate
    - [ ] RPC methods?
    - [ ] Chainspec, both JSON specs & the module?
    - [ ] Runtime versioning?
    - [ ] Benchmarks?
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Feel free to explain in further detail your choices if this is a significant change.

### CODE REVIEW

File: client/packages/cli/index.ts [hunk 1]
```diff
@@ -159,11 +211,11 @@ class CircuitCLI {
             // @ts-ignore
             let submissionHeight = await this.sdk.circuit.tx.signAndSendSafe(tx)
                 .then(height => {
-                    console.log("Success: Transfer submitted!")
+                    logger.info("Success: Transfer submitted!")
                     return height
                 })
                 .catch(err => {
-                    console.error("Error: Transfer submission failed! Err:", err);
+                    logger.info("Error: Transfer submission failed! Err:", err);
```
Comment (3h4x) [inline client/packages/cli/index.ts:218]: logger.info or logger.error?

File: client/packages/cli/index.ts [hunk 2]
```diff
@@ -191,7 +243,7 @@ class CircuitCLI {
         // Check we have an config for each SideEffect
         data.sideEffects.forEach(effect => {
             if (!config.gateways.find(entry => entry.id === effect.target)) {
-                console.error(`Error: Gateway for SideEffect ${effect.type} not found!`)
+                logger.info(`Error: Gateway for SideEffect ${effect.type} not found!`)
```
Comment (3h4x) [inline client/packages/cli/index.ts:246]: info or error?

Comment (3h4x) [inline client/packages/cli/index.ts:246]: could have extra kwarg {sideeffect: effect.type)
this way we could see and visualise easily which are not found

File: client/packages/cli/index.ts [hunk 3]
```diff
@@ -201,11 +253,11 @@ class CircuitCLI {
 
         const submissionHeight = await this.sdk.circuit.tx.signAndSendSafe(tx)
             .then(height => {
-                console.log("Success: SideEffects submitted!!")
+                logger.info("Success: SideEffects submitted!!")
                 return height
             })
             .catch(err => {
-                console.error("Error: SideEffects submission failed! Err:", err);
+                logger.info("Error: SideEffects submission failed! Err:", err);
```
Comment (3h4x) [inline client/packages/cli/index.ts:260]: info or error?

File: client/packages/cli/index.ts [hunk 4]
```diff
@@ -201,11 +253,11 @@ class CircuitCLI {
 
         const submissionHeight = await this.sdk.circuit.tx.signAndSendSafe(tx)
             .then(height => {
-                console.log("Success: SideEffects submitted!!")
+                logger.info("Success: SideEffects submitted!!")
```
Comment (3h4x) [inline client/packages/cli/index.ts:256]: success not needed, one exclamation mark

**Review by 3h4x — COMMENTED** [review]
Good work. Those logs will be useful.
I got a couple comments where adding prefix Error: or Success: is not providing us searchable information.
It's better to have loglevel and pass error, info, etc there. That way, we can exclude Info logs and browse everything else, which will be also easy on our log storage IO.

File: client/packages/cli/index.ts (hunk 2 above)
Comment (petscheit) [reply client/packages/cli/index.ts:246]: Ahaig, this is user facing logs, so no field formatting is needed

File: client/packages/cli/index.ts (hunk 1 above)
Comment (3h4x) [reply client/packages/cli/index.ts:218]: Okay but the prefix `Error:` is not needed. We already communicate that transfer failed.
Err should be in logging kwargs

**Review by MaciejBaj — APPROVED** [review]

**Pull request merged**